File: APIdoc/scala/apidoc/src/main/scala/io/point/tutorials/ListFoldersDownloadFile.scala
package io.point.tutorials

import io.point.Pio
import dispatch._, Defaults._
import play.api.libs.json.JsArray

/**
  * User: jconley
  * Date: 5/7/2014
  */
object ListFoldersDownloadFile {
   def main(args: Array[String]) {
     val (email, password, apiKey) = ("", "", "")

     for {
       keyJson <- Pio.authenticate(email, password, apiKey)
       sessionKey = keyJson.getOrElse(throw new Exception("Session key not found"))
       accessRulesJson <- Pio.listAccessRules(sessionKey)
       accessRules = accessRulesJson.getOrElse(throw new Exception("Access rules not found"))
       foldersJson <- Pio.listFolders(sessionKey, accessRules(2).as[JsArray].value(1).as[String])
       folders = foldersJson.getOrElse(throw new Exception("Folders not found"))
     } yield {
       folders.value.foreach{ f =>
         val folder = f.as[JsArray]
         if(folder.value(2).as[String] == "FILE"){
           Pio.fileDownload(sessionKey, folder.value(16).as[String], folder.value(0).as[String], folder.value(1).as[String]).map{ res =>
             println(res)
           }
         }
       }
     }
   }
 }

File: APIdoc/c-sharp/ListFoldersDownloadFile.cs
using System;
using System.Net;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Web.Script.Serialization;

namespace Pio{
    public class ListFoldersDownloadFile{
        public static void Main(String[] args) {

            if(args.Length != 3){
                Console.WriteLine("Invalid args, should be email password apiKey");
                Environment.Exit(-1);
            }

            String sessionKey = APIDoc.authenticate(args[0], args[1], args[2]);
            
            ArrayList rules = APIDoc.listAccessRules(sessionKey);
            ArrayList rule = rules[2] as ArrayList;
            String folderId = rule[1] as String;

            ArrayList folders = APIDoc.listFolders(sessionKey, folderId);
            ArrayList file = folders[5] as ArrayList;

            String downloadUrl = fileDownload(sessionKey, file[16] as string, file[0] as string, file[1] as string);
            Console.WriteLine(downloadUrl);
        }
    }
}